### PR TITLE
VAM-145: XUS PKI GET UPN

### PR DESCRIPTION
**Pull request by creindle**
[X] Passes comparison tests
[X] Passes model tests

### CODE REVIEW

File: emulation/models/xus-pki-get-upn.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        files: ['200'],
+        parameters: [],
+        labels: ['META'],
+        services: ['utilityMetaService'],
+        mvdm: {
+            New_Person: ['subjectAlternativeName'],
+        },
+        inputParameters: [],
+        invokeRPC: (rpcArgs, session, callback) => {
+            utilityMetaService.subjAltNameField(session, callback);
```
Comment (mfuroyama) [inline emulation/models/xus-pki-get-upn.js]: Let's make the name of the service function a little more descriptive and get it to conform to the ["VAM Commandments"](https://github.com/vistadataproject/VICSServer/wiki/Guidelines-for-VICS-Services)

Consider a function name like, `lookupUserAlternativeName` or something.

And, even though the RPC domain is listed as `'UTILITY'` let's also consider moving the service function from `utilityMetaService` over to `userMetaService`. It's a subtle difference, but I think it's a better logical grouping. Let's discuss...

File: services/utilityMetaService/lookupSubjAltNameField.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * @summary This RPC gets the SUBJECT ALTERNATIVE NAME field from the New Person (#200) file field 501.2. It is used to check that the correct PIV card has been put into the reader.
+ * @param  {Object}   context  Session context information associated with the service call.
+ * @param  {Function} callback Function called on process completion, with standard (err, result) signature.
+ * @return {String}   subjectAlternativeName value from response or empty string
+ *
+ * @memberof UtilityMetaService
+ * @instance
+ */
+function subjAltNameField(context, callback) {
```
Comment (mfuroyama) [inline services/utilityMetaService/lookupSubjAltNameField.js]: See my comment above about possibly renaming this function...

File: emulation/models/xus-pki-get-upn.js (hunk 1 above)
Comment (creindle) [reply emulation/models/xus-pki-get-upn.js]: Oh no! I changed the file name and forgot to change the service function name too! Also, yes I would like to discuss the change from `utilityMetaService` to `userMetaService` so I understand more of what goes in each service group.

File: emulation/models/xus-pki-get-upn.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        function: 'GETUPN^XUSER2',
+        domain: 'UTILITY',
+        description: 'This RPC gets the SUBJECT ALTERNATIVE NAME field from the New Person (#200) file field 501.2. It is used to check that the correct PIV card has been put into the reader.',
+        rpcType: 'READ',
+        complexity: 'LOW',
+        returnType: 'SINGLE VALUE',
+        files: ['200'],
+        parameters: [],
+        labels: ['META'],
+        services: ['utilityMetaService'],
```
Comment (creindle) [inline emulation/models/xus-pki-get-upn.js]: So would I add 'userMetaService' here? Or replace `utilityMetaService`?

File: emulation/models/xus-pki-get-upn.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        function: 'GETUPN^XUSER2',
+        domain: 'UTILITY',
+        description: 'This RPC gets the SUBJECT ALTERNATIVE NAME field from the New Person (#200) file field 501.2. It is used to check that the correct PIV card has been put into the reader.',
+        rpcType: 'READ',
+        complexity: 'LOW',
+        returnType: 'SINGLE VALUE',
+        files: ['200'],
+        parameters: [],
+        labels: ['META'],
+        services: ['userMetaService', 'userMetaService'],
```
Comment (mfuroyama) [inline emulation/models/xus-pki-get-upn.js]: Hmm... 🤔  possibly one too many `'userMetaService'` entries listed here...

File: services/userMetaService/lookupSubjAltNameField.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * @summary This RPC gets the SUBJECT ALTERNATIVE NAME field from the New Person (#200) file field 501.2. It is used to check that the correct PIV card has been put into the reader.
+ * @param  {Object}   context  Session context information associated with the service call.
+ * @param  {Function} callback Function called on process completion, with standard (err, result) signature.
+ * @return {String}   subjectAlternativeName value from response or empty string
+ *
+ * @memberof UtilityMetaService
+ * @instance
+ */
+function lookupSubjAltNameField(context, callback) {
```
Comment (mfuroyama) [inline services/userMetaService/lookupSubjAltNameField.js]: Yeah...you know what? This is somewhat pedantic, but let's change the function name here to `lookupAlternativeName`. For published API functions, it's **usually** a good practice to leave out abbreviations when naming functions.

Comment (creindle) [reply services/userMetaService/lookupSubjAltNameField.js]: I never realized that! So very good to know the best practices!

File: emulation/models/xus-pki-get-upn.js (hunk 4 above)
Comment (creindle) [reply emulation/models/xus-pki-get-upn.js]: I apologize!

**Pull request merged**